plugins/by-name/treesitter/default.nix

```
@@ -366,24 +232,80 @@ lib.nixvim.plugins.mkNeovimPlugin {
   callSetup = false;
 
   extraConfig = cfg: opt: {
+    # TODO: Added 2025-12-18 Runtime detection to support legacy api
+    # Should be removed after transition period
+    # Runtime API detection and conditional setup
     plugins.treesitter.luaConfig.content =
-      # NOTE: Upstream state that the parser MUST be at the beginning of runtimepath.
-      # Otherwise the parsers from Neovim takes precedent, which may be incompatible with some queries.
-      (optionalString (cfg.settings.parser_install_dir != null) ''
-        vim.opt.runtimepath:prepend(${lib.nixvim.toLuaObject cfg.settings.parser_install_dir})
-      '')
-      + ''
-        require('nvim-treesitter.configs').setup(${lib.nixvim.toLuaObject cfg.settings})
+      let
+        # TODO: Added 2025-12-18 Check both legacy and new api options
+        # Add warning after transition period
+        # Map parser_install_dir to install_dir for main branch compatibility
+        # The main branch uses install_dir, while master branch uses parser_install_dir
+        mainBranchSettings =
+          lib.optionalAttrs ((cfg.settings.parser_install_dir or null) != null) {
```

Do we actually want to explicitly skip `null`, or is the intent:

```suggestion
          lib.optionalAttrs (cfg.settings ? parser_install_dir) {
```

...matching the `? install_dir` check below?